src/de/grobox/liberario/fragments/DirectionsFragment.java

```diff
@@ -426,9 +435,42 @@ private void processIntent() {
 		if(intent != null) {
 			final String action = intent.getAction();
 			if(action != null && action.equals(TAG)) {
-				Location from = (Location) intent.getSerializableExtra("from");
-				Location via = (Location) intent.getSerializableExtra("via");
-				Location to = (Location) intent.getSerializableExtra("to");
+				WrapLocation wfrom = (WrapLocation) intent.getSerializableExtra("from");
+				WrapLocation wvia = (WrapLocation) intent.getSerializableExtra("via");
+				WrapLocation wto = (WrapLocation) intent.getSerializableExtra("to");
+				Location from, via, to;
+				if(wfrom != null) {
+					from = wfrom.getLocation();
+					if(wfrom.getType() == WrapLocation.WrapType.GPS) {
+						activateGPS();
+						from = null;
+					}
+				} else {
+					from = null;
+				}
+				if(wvia != null) {
+					via = wvia.getLocation();
+				} else {
+					via = null;
+				}
+				if(wto != null) {
+					to = wto.getLocation();
+					if(wto.getType() == WrapLocation.WrapType.HOME){
+						Location home = RecentsDB.getHome(getContext());
+						if(home != null) {
+							to = home;
+						} else {
+							// show home picker dialog
+							changingHome = true;
+							HomePickerDialogFragment setHomeFragment = HomePickerDialogFragment.newInstance();
+							setHomeFragment.setOnHomeChangedListener(this);
+							FragmentTransaction ft = getActivity().getSupportFragmentManager().beginTransaction();
+							setHomeFragment.show(ft, HomePickerDialogFragment.TAG);
```

Comment:
The indentation is getting crazy here. Maybe this whole part get be factored out into two methods. one that processes the intent and the other showing the HomePickerDialogFragment if necessary. But I am not sure, we need this HomePickerDialogFragment stuff here.

A first idea is to maybe have a `LocationView#setLocation()` method that takes directly a `WrapLocation` and then handles everything based on those wrap locations on its own like it is now.